lib/services/role.go

```diff
@@ -2243,6 +2243,12 @@ func NewLoginMatcher(login string) RoleMatcher {
 
 // Match matches a login against a role.
 func (l *loginMatcher) Match(role types.Role, typ types.RoleConditionType) (bool, error) {
+	// When joining sessions, we can't match on the OS login.
```

Comment:
While this allows the login matcher to succeed for `-teleport-internal-join`, the node matcher is still checked and fails.